src/components/Discover.js

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect } from "react";
+
+function Discover(props) {
+    
+    useEffect(() => {
+        function handleChange (e) {
+            props.setSelectedDate(e.target.value);
+            props.setnasaAPI(`https://api.nasa.gov/planetary/apod?api_key=krL1cCbGBXpZ5c5cti1GnKp60Z8ueT8RqpypcSlY&date=${e.target.value}`)
+        }
+        const dateInput = document.querySelector('input')
+        dateInput.addEventListener('change', handleChange)
+
```

Comment:
With React, you don't need to do this the traditional way, adding the `onChange` event on the input can cater for it.
`<input type="date" className="date"  onChange={handleChange} />`